src/System.Private.CoreLib/shared/System/Text/ValueStringBuilder.cs

```diff
@@ -171,8 +171,10 @@ public void Append(char c)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public void Append(string s)
+        public void Append(string? s)
         {
+            if (s is null) return;
```

Comment:
Does enough places pass null into this to make this worth it?

Reply:
(This is aggressive inlined everywhere.)

Reply:
`.ToString` on object returning a `string?` doesn't help :) But no, putting the guards in the call sites.

Reply:
> .ToString on object returning a string? doesn't help :)

Other than in advanced implementations like Format using this, where is ToString being passed to Append on types that don't override it?

Reply:
`.ToString` in `JoinCore<T>` on `T`; `.ToString` in `JoinCore` on `object` etc changes in https://github.com/dotnet/coreclr/pull/26934/commits/b2322288da0814fe14deb5ec88b1e98e48b32f86

Are basically on `object` or `T`

Reply:
Right, that's what I meant by advanced implementations, helpers in core types like string itself. And such methods need to check for null, anyway. We've had bugs filed in the past because sometimes they haven't, e.g. 
https://github.com/dotnet/coreclr/issues/6785
https://github.com/dotnet/roslyn/issues/34085

Reply:
There is a lot of places that need the null check. I am thinking it would be better to just keep it inside the method. @benaadams @stephentoub Do you have an opinion?

Reply:
I'd prefer it inside:
- Better matches String builder
- Easier to consume
- Doesn't add a lot of complexity
- Will likely lead to less code due to number of call sites that needed the check